models/vrp.rb

```diff
@@ -350,9 +362,9 @@ def self.generate_schedule_indices_from_date(hash)
 
       # convert route dates into indices
       hash[:routes]&.each{ |route|
-        next if route[:indice]
+        next if route[:indice] || route[:index]
```

Comment:
If the retrocompatibility function is called before this function then we don't need the `route[:indice] ||` part. 

If not, we should move the `convert_route_indice_into_index` earlier. 

After a parameter is deprecated, it should not appear anywhere in the API except the location where we handle "switch"